internal/extender/resourcereservations.go

```diff
@@ -191,8 +216,86 @@ func (rrm *ResourceReservationManager) GetReservedResources() resources.NodeGrou
 	return usage
 }
 
+// CompactDynamicAllocationApplications compacts reservations for executors belonging to dynamic allocation applications by moving
+// any soft reservations to resource reservations occupied by now-dead executors. This ensures we have relatively up to date resource
+// reservation objects and report correctly on reserved usage.
+func (rrm *ResourceReservationManager) CompactDynamicAllocationApplications(ctx context.Context) {
+	timer := metrics.GetAndStartSoftReservationCompactionTimer()
+	defer timer.MarkCompactionComplete(ctx)
+	dynamicAllocationAppsToCompact := rrm.drainDynamicAllocationCompactionSlice()
+
+	rrm.mutex.Lock()
+	defer rrm.mutex.Unlock()
+	for _, app := range dynamicAllocationAppsToCompact {
+		if sr, ok := rrm.softReservationStore.GetSoftReservation(app.appID); ok {
+			rrm.logger.Info("starting executor compaction for application", svc1log.SafeParam("appID", app.appID))
+			pods, err := rrm.getActivePods(ctx, app.namespace, app.appID)
+			if err != nil {
+				rrm.logger.Error("error getting active pods during compaction", svc1log.SafeParam("podNamespace", app.namespace), svc1log.Stacktrace(err))
+				continue
+			}
+			for podName := range sr.Reservations {
+				pod, ok := pods[podName]
+				if !ok {
+					rrm.logger.Info("executor pod with soft reservation no longer active, skipping compaction for this one",
+						svc1log.SafeParam("podNamespace", pod.Namespace),
```

Comment:
these will clash with the top level params we add in the `Predicate` function, we should rename them to something like`podNamespaceUnderCompation`

Reply:
Discussed and decided to keep as is since it will not clash given we're using the `ResourceReservationManager` logger which we construct with another context